recognition/src/face_detection/face_detector_data_provider.cpp

```diff
@@ -282,15 +282,15 @@ void pcl::face_detection::FaceDetectorDataProvider<FeatureType, DataSet, LabelTy
     if (USE_NORMALS_)
     {
       integral_image_normal_x.reset (new pcl::IntegralImage2D<float, 1> (false));
-      const float *data_nx = reinterpret_cast<const float*> (&(*normals)[0]);
+      const float *data_nx = reinterpret_cast<const float*> (&(*normals)[0]);// NOLINT(readability-container-data-pointer)
```

Comment:
I guess one could just:
`const float* data = reinterpret_cast<const float*> (*normals).data();`

And in the two other, with y and z, have the call
`setInput(data + 1, ....)  `- for Y
and
`setInput(data + 2, ...)` - for Z

I'm not entirely sure what should be preferred, but currently all 3 statements return the same pointer, but calls them data_nx, data_ny and data_nz - which seems confusing to me 😄